plugins/UsersManager/API.php

```diff
@@ -790,6 +844,16 @@ public function getTokenAuth($userLogin, $md5Password)
     {
         UsersManager::checkPasswordHash($md5Password, Piwik::translate('UsersManager_ExceptionPasswordMD5HashExpected'));
 
-        return md5($userLogin . $md5Password);
+        $user = $this->model->getUser($userLogin);
+
+        if (!$this->password->verify($md5Password, $user['password'])) {
+            return hash('sha256', $userLogin . microtime(true) . Common::generateUniqId());
```

Comment:
Can we maybe move the `hash('sha256'...` into `Password` class as well? I know it doesn't make much sense to put there from the naming perspective cause Password != Token but be nice to have it always in one place :)